src/properties/degrees.c

```diff
@@ -517,3 +517,138 @@ igraph_error_t igraph_sort_vertex_ids_by_degree(const igraph_t *graph,
+                              IGRAPH_EINVAL,
+                              dout, din, max_out_degree, max_in_degree);
+            }
+            IGRAPH_CHECK(igraph_matrix_int_resize(jdm, dout, din));
+        } else {
+            IGRAPH_CHECK(igraph_matrix_int_resize(jdm, max_out_degree, max_in_degree));
+        }
+        // Set all elements to 0
+        igraph_matrix_int_null(jdm);
+
+        IGRAPH_CHECK(igraph_es_all(&es, IGRAPH_EDGEORDER_ID));
+        IGRAPH_CHECK(igraph_eit_create(graph, es, &eit));
+
+        while (!IGRAPH_EIT_END(eit)) {
+            eid = IGRAPH_EIT_GET(eit);
+            v1id = IGRAPH_FROM(graph, eid);
+            v2id = IGRAPH_TO(graph, eid);
+            v1deg = igraph_vector_int_get(&out_degrees,v1id);
+            v2deg = igraph_vector_int_get(&in_degrees, v2id);
+            if (!weights) {
+                MATRIX(*jdm, v1deg-1, v2deg-1)++;
+            } else {
+                MATRIX(*jdm, v1deg-1, v2deg-1) += VECTOR(*weights)[eid];
+            }
+            IGRAPH_EIT_NEXT(eit);
+        }
+
+        igraph_vector_int_destroy(&out_degrees);
+        igraph_vector_int_destroy(&in_degrees);
+
+    } else {
+        igraph_vector_int_t degrees;
+        igraph_integer_t max_degree;
+        // Compute max degrees
+        IGRAPH_CHECK(igraph_vector_int_init(&degrees, 0));
+        IGRAPH_CHECK(igraph_degree(graph, &degrees, igraph_vss_all(), IGRAPH_ALL, true));
+        IGRAPH_CHECK(igraph_maxdegree(graph, &max_degree, igraph_vss_all(), IGRAPH_ALL, true));
+        if (dout > 0 && din > 0) {
+            if (dout < max_degree || din < max_degree) {
+                IGRAPH_ERRORF("dout (%" IGRAPH_PRId ") or din (%" IGRAPH_PRId ") are smaller than the smallest required dimensions of the matrix (%" IGRAPH_PRId ", %" IGRAPH_PRId ")",
+                              IGRAPH_EINVAL,
+                              dout, din, max_degree, max_degree);
+            }
+            IGRAPH_CHECK(igraph_matrix_int_resize(jdm, dout, din));
+        } else {
+            IGRAPH_CHECK(igraph_matrix_int_resize(jdm, max_degree, max_degree));
+        }
+        igraph_matrix_int_null(jdm);
+        IGRAPH_CHECK(igraph_es_all(&es, IGRAPH_EDGEORDER_ID));
+        IGRAPH_CHECK(igraph_eit_create(graph, es, &eit));
+
+        while (!IGRAPH_EIT_END(eit)) {
+            eid = IGRAPH_EIT_GET(eit);
+            v1id = IGRAPH_FROM(graph, eid);
+            v2id = IGRAPH_TO(graph, eid);
+            v1deg = igraph_vector_int_get(&degrees, v1id);
+            v2deg = igraph_vector_int_get(&degrees, v2id);
+            if (!weights) {
+                MATRIX(*jdm, v1deg - 1, v2deg - 1)++;
+                MATRIX(*jdm, v2deg - 1, v1deg - 1)++;
+            } else {
+                MATRIX(*jdm, v1deg - 1, v2deg - 1) += VECTOR(*weights)[eid];
+                MATRIX(*jdm, v2deg - 1, v1deg - 1) += VECTOR(*weights)[eid];
+            }
+
+            IGRAPH_EIT_NEXT(eit);
+        }
+
+        igraph_vector_int_destroy(&degrees);
```

Comment:
```suggestion
        igraph_vector_int_destroy(&degrees);
        IGRAPH_FINALLY_CLEAN(1);
```

This de-registers the last data structure we registered for de-allocation. We need to do this since we de-allocated it ourselves here.

If we have `n` instead of `1` as the argument, it would de-register the last `n` instead of the last one.